[TG-7618] Fix precision parsing in string_format

romainbrenguier opened the pull request:
The 4th group in the regular expression matches `(\.\d+)?` so the dot
should be ignored before the string can be parsed to an int.

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [na] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [na] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [na] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

romainbrenguier committed:
Fix precision parsing in string_format

The 4th group in the regular expression matches `(\.\d+)?` so the dot
should be ignored before the string can be parsed to an int.

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 9c7b44a48620c378de62cd09cfa720e9ad571a6b).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/111548705

tautschnig commented:
Would it be possible to include a test?

martin-cs reviewed (approved):
Sounds believable but a "fails before but works now" test case, either system or unit, would be good.

romainbrenguier committed:
Add test for String.format with float

This checks that having a format flag with precision defined doesn't
crash the solver.

romainbrenguier commented:
I started adding a test but it will require a model for Float in java-models-library

romainbrenguier commented:
Outdated
